### PR TITLE
[#1837562] Migrate function app nodes in azure explorer to app service library

### PR DESCRIPTION
**Pull request by Flanker32**
- Refactor web app base node with app service library
- Migrate function app nodes in azure explorer to app service library

### CODE REVIEW

**Comment by Flanker32** [comment]
AB#1837562

File: Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/function/FunctionAppNode.java [hunk 1]
```diff
@@ -94,28 +91,25 @@ public String getFunctionAppName() {
...
-        return this.functionApp.regionName();
+        return this.functionApp.entity().getRegion().getName();
     }
 
     @AzureOperation(name = "function.start", params = {"this.functionApp.name()"}, type = AzureOperation.Type.ACTION)
     private void start() {
-        AzureFunctionMvpModel.getInstance().startFunction(subscriptionId, this.getFunctionAppId());
-        FunctionApp target = AzureFunctionMvpModel.getInstance().getFunctionById(subscriptionId, this.getFunctionAppId());
-        this.renderNode(WebAppBaseState.fromString(target.state()));
+        Azure.az(AzureAppService.class).functionApp(id).start();
```
Comment (Flanker32) [inline Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/function/FunctionAppNode.java:99]: replace with this.functionApp

File: Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/base/WebAppBaseState.java [hunk 2]
```diff
@@ -8,7 +8,8 @@
 public enum WebAppBaseState {
     RUNNING,
```
Comment (Flanker32) [inline Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/base/WebAppBaseState.java:9]: todo: replace with string constant

File: Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/base/WebAppBaseState.java [hunk 3]
```diff
@@ -8,7 +8,8 @@
...
     STOPPED,
-    UPDATING;
```
Comment (Flanker32) [inline Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/base/WebAppBaseState.java:11]: todo: move state enum to lib

**Review by andxu — APPROVED** [review]

**Pull request merged**